### PR TITLE
Add `[lang]-query-file` arg

### PR DESCRIPTION
**Pull request by bheylin**
Polish off #144

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,7 +148,7 @@ some combinations from making sense: for example, searching for a Python `class`
 however:
 
 ```console
-$ cat birds.py | srgn --py 'class' --py 'doc-strings' # From earlier example; note `--py` is a shorthand for `--python`
```
Comment (alexpovel) [inline README.md:151]: This stopped working due to 

https://github.com/alexpovel/srgn/pull/168/files#diff-c4682009f01c1c630ec1481e9dae8b61781728c8aa12f1f9000abea84f1cf810R522-R524

?

Can you remove unused new entries such as `tag("rust-query-file"),`, then comment in `tag("py")` etc. again? That way we get both worlds working (not touching existing docs), at the cost of not having 1:1 compatibility between `readme.rs` and the actual CLI. Doesn't seem like a high price to pay (implementers of new README examples might run into bizarre errors... I'll document the "here be dragons" part of this code base better)

**Review by alexpovel — COMMENTED** [review]
LGTM!

**Comment by alexpovel** [comment]
CI failing at https://github.com/alexpovel/srgn/actions/runs/11693746764/job/32611046155?pr=168#step:7:321 because some GitHub Actions runners we get are on Rust 1.82, some on 1.81. Flaky! But irrelevant. Will maybe fix on `main`.

**Comment by alexpovel** [comment]
If you rebase onto `main`, there's some goodies waiting:

- 2ce5a1920ffe82b58ebb6b1caeb0dd8b145ab979 lets you run `./scripts/update-readme.py` and it'll update the `srgn --help` section automatically (not tested extensively yet)
- bdd183cc27036f7840db2c0d85a89a7d04c0dd0e will let you add as many `tag()` as you want. Your new arguments are the longest of all existing ones, so put them first:

    https://github.com/alexpovel/srgn/blob/bdd183cc27036f7840db2c0d85a89a7d04c0dd0e/tests/readme.rs#L495-L503
    
    Put them in their own `arg` group like the others and it should fit and work.
- 3af3f52c428a526e15239b1759f8500080fbba0a removed the bug required the hack with a trailing space
- 0e100dd0fc4798672ffe880d5654ca23387aefbb silences the lint which is failing CI

File: README.md (hunk 1 above)
Comment (bheylin) [reply README.md:151]: I reverted the doc changes in cdc04bc702931e1dfbf8dd1f16756433ecdbcc89 and added all `--[lang-query-file]` args to the `readme.rs` parser in an `alt` combinator like the rest of the args in a77b149274f70f288f21f490464478f751e6af5b.

**Comment by bheylin** [comment]
> If you rebase onto `main`, there's some goodies waiting:
> 
>     * [2ce5a19](https://github.com/alexpovel/srgn/commit/2ce5a1920ffe82b58ebb6b1caeb0dd8b145ab979) lets you run `./scripts/update-readme.py` and it'll update the `srgn --help` section automatically (not tested extensively yet)

Ah good quality of life improvement!

> 
>     * [bdd183c](https://github.com/alexpovel/srgn/commit/bdd183cc27036f7840db2c0d85a89a7d04c0dd0e) will let you add as many `tag()` as you want. Your new arguments are the longest of all existing ones, so put them first:
>       https://github.com/alexpovel/srgn/blob/bdd183cc27036f7840db2c0d85a89a7d04c0dd0e/tests/readme.rs#L495-L503
>       
>       Put them in their own `arg` group like the others and it should fit and work.
> 

Done! a77b149274f70f288f21f490464478f751e6af5b.

>     * [3af3f52](https://github.com/alexpovel/srgn/commit/3af3f52c428a526e15239b1759f8500080fbba0a) removed the bug required the hack with a trailing space
> 

Good to get that confusing issue fixed.

>     * [0e100dd](https://github.com/alexpovel/srgn/commit/0e100dd0fc4798672ffe880d5654ca23387aefbb) silences the lint which is failing CI

:partying_face:

**Comment by alexpovel** [comment]
Great, merged! Thanks for your work, I'll release soon and then you can finally use it.

**Comment by bheylin** [comment]
> Great, merged! Thanks for your work, I'll release soon and then you can finally use it.

Nice! I'll be trying it out as part of the CI process of a 200K LOC Rust project to perform custom linting.
I'll be back with feedback and more PR's!

**Pull request merged**